File: code/dotnet/src/Examples/GoogleTest.cs
using System;
using NUnit.Framework;
using Selenium;
using ThoughtWorks.Selenium.Core;

namespace ThoughtWorks.Selenium.IntegrationTests
{
	[TestFixture]
	public class GoogleTest
	{
		private ISelenium selenium;

		[SetUp]
		public void SetupTest()
		{
			HttpCommandProcessor processor = new HttpCommandProcessor();
			DefaultBrowserLauncher launcher = new InternetExplorerBrowserLauncher();
			selenium = new DefaultSelenium(processor, launcher, DefaultSelenium.SELENESE_RUNNER_URL.Replace("html", "hta"));
			selenium.Start();
		}

		[TearDown]
		public void TeardownTest()
		{
			try
			{
				selenium.Stop();
			}
			catch (Exception)
			{
				// Ignore errors if unable to close the browser
			}
		}

		[Test]
		public void GoogleSearch()
		{
			selenium.Open("http://www.google.com");
			selenium.VerifyTitle("Google");
			/*
			selenium.Type("q", "Selenium ThoughtWorks");
			selenium.VerifyValue("q", "Selenium ThoughtWorks");
			selenium.Click("btnG");
			selenium.VerifyTextPresent("", "selenium.thoughtworks.com");
			selenium.VerifyTitle("Google Search: Selenium ThoughtWorks");
			*/
			selenium.TestComplete();
		}
	}
}

File: clients/php/GoogleTest.php
<?php

set_include_path(get_include_path() . PATH_SEPARATOR . './PEAR/');
require_once 'Testing/Selenium.php';
require_once 'PHPUnit/Framework/TestCase.php';

class GoogleTest extends PHPUnit_Framework_TestCase
{
    private $selenium;

    public function setUp()
    {
        $this->selenium = new Testing_Selenium("*firefox", "http://www.google.com");
        $this->selenium->start();
    }

    public function tearDown()
    {
        $this->selenium->stop();
    }

    public function testGoogle()
    {
        $this->selenium->open("/");
        $this->selenium->type("q", "hello world");
        $this->selenium->click("btnG");
        $this->selenium->waitForPageToLoad(10000);
        $this->assertRegExp("/Google Search/", $this->selenium->getTitle());
    }

}
?>
